clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/AbstractPartitionStateModel.java

```diff
@@ -352,8 +352,22 @@ private void initializePartitionPushStatus() {
     }
   }
 
-  protected void stopConsumption() {
-    ingestionBackend.stopConsumption(storeAndServerConfigs, partition);
+  protected void stopConsumption(boolean dropCVState) {
+    CompletableFuture<Void> future = ingestionBackend.stopConsumption(storeAndServerConfigs, partition);
+    if (dropCVState) {
+      future.whenComplete((ignored, throwable) -> {
```

Comment:
Why wait until after the consumption is stopped? Why not remove the CV first, then initiate the stop of consumption? Would we not want to notify routers about the CV drop ASAP? In the end, it'll be a race condition anyway whether the router finds out about it before the consumption is stopped or not, but it seems more natural to me to begin notifying routers first since we don't intend them to keep sending traffic here...

Reply:
The reason to have such logic is that `stopConsumption` is async today.
If we update CV state first here, the CV state can be changed again when executing `stopConsumption` logic in `SIT`.
That was the main reason to have such callback.

Reply:
Why will the CV state be changed again? Is it because of this:

```
statusReportAdapter.reportStopped(consumptionState);
```

In `SIT::processCommonConsumerAction`?

Reply:
Well, when trying to stop consumption, `SIT` will try to drain the queued messages, and when drainer processing these messages, it can change the replica state, such as changing it to be `completed` or `errored` , which means CV state will be updated after removal.

Reply:
Hmm... this is a bigger scope of change and it is fine to ignore it in this PR, but I wonder... what is the point of this draining? Why not just stop instantly, and later on skip any message belonging to an unsubscribed partition, rather than do this draining?

Reply:
Yeah, we had such conversion about making ST sync to make logic more straightforward.
Today, handling these consumer actions are completely async.
It will be a much bigger change and I would like to keep that as a separate PR.